components/support/rs_client/src/http_client.rs

```diff
@@ -0,0 +1,226 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! This is a simple HTTP client that uses viaduct to retrieve Remote Settings data from the server.
+
+use std::time::{Duration, Instant};
+
+use crate::config::ClientConfig;
+use crate::error::{ClientError, Result};
+use std::cell::Cell;
+use url::Url;
+use viaduct::{status_codes, Request, Response};
+
+const HEADER_BACKOFF: &str = "Backoff";
+const HEADER_RETRY_AFTER: &str = "Retry-After";
+
+pub struct Client {
+    pub(crate) base_url: Url,
+    pub(crate) bucket_name: String,
+    pub(crate) collection_name: String,
+    pub(crate) remote_state: Cell<RemoteState>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) enum RemoteState {
+    Ok,
+    Backoff {
+        observed_at: Instant,
+        duration: Duration,
+    },
+}
+
+impl Client {
+    pub fn new(config: ClientConfig) -> Result<Self> {
+        let server_url = config
+            .server_url
+            .unwrap_or_else(|| String::from("https://firefox.settings.services.mozilla.com"));
+        let bucket_name = config.bucket_name.unwrap_or_else(|| String::from("main"));
+
+        Ok(Self {
+            base_url: Url::parse(&server_url)?,
+            bucket_name,
+            collection_name: config.collection_name,
+            remote_state: Cell::new(RemoteState::Ok),
+        })
+    }
+
+    pub fn get(&self) -> Result<Response> {
```

Comment:
how do you feel about the name `get`, maybe `get_records` is clearer? I'm not sure if it's easy to imply by consumers that `get` refers to the records but I don't have too much context on remote_settings as a whole 😅

Reply:
Maybe I could add something here, how do you feel about adding doc comments to the `http_client` functions here? Since it's its own library now, it could be useful for other Rust consumers to see the docs for each method! ref: https://doc.rust-lang.org/rust-by-example/meta/doc.html

Reply:
I'm not at all married to the name, and I agree that `get_records` is clearer. This was intended to match with the [existing platform client](https://remote-settings.readthedocs.io/en/latest/introduction.html#what-does-the-client-api-look-like), but if we even deem that to be important it could be done in the top-level Remote Settings client instead of here.